plugins/modules/system/sapcontrol.py

```diff
@@ -0,0 +1,389 @@
+               'ABAPGetWPTable', 'ABAPGetComponentList', 'ABAPCheckRFCDestinations',
+               'ABAPGetSystemWPTable', 'J2EEControlProcess', 'J2EEControlCluster', 'J2EEEnableDbgSession',
+               'J2EEDisableDbgSession', 'J2EEGetProcessList', 'J2EEGetProcessList2', 'J2EEGetThreadList', 'J2EEGetThreadList2',
+               'J2EEGetThreadCallStack', 'J2EEGetThreadTaskStack', 'J2EEGetSessionList', 'J2EEGetCacheStatistic',
+               'J2EEGetCacheStatistic2', 'J2EEGetApplicationAliasList', 'J2EEGetComponentList',
+               'J2EEControlComponents', 'J2EEGetWebSessionList', 'J2EEGetWebSessionList2', 'J2EEGetEJBSessionList', 'J2EEGetRemoteObjectList',
+               'J2EEGetVMGCHistory', 'J2EEGetVMGCHistory2', 'J2EEGetVMHeapInfo', 'J2EEGetClusterMsgList', 'J2EEGetSharedTableInfo',
+               'ICMGetThreadList', 'ICMGetConnectionList', 'ICMGetProxyConnectionList', 'ICMGetCacheEntries', 'WebDispGetServerList',
+               'WebDispGetGroupList', 'WebDispGetVirtHostList', 'WebDispGetUrlPrefixList', 'EnqGetStatistic', 'EnqGetLockTable',
+               'EnqRemoveUserLocks', 'StartWait', 'StopWait', 'WaitforStarted', 'WaitforStopped', 'RestartServiceWait',
+               'WaitforServiceStarted', 'CheckHostAgent']
+    return retlist
+
+
+# converts recursively the suds object to a dictionary e.g. {'item': [{'name': hdbdaemon, 'value': '1'}]}
+def recursive_dict(suds_object):
+    out = {}
+    if isinstance(suds_object, str):
+        return suds_object
+    for k, v in asdict(suds_object).items():
+        if hasattr(v, '__keylist__'):
+            out[k] = recursive_dict(v)
+        elif isinstance(v, list):
+            out[k] = []
+            for item in v:
+                if hasattr(item, '__keylist__'):
+                    out[k].append(recursive_dict(item))
+                else:
+                    out[k].append(item)
+        else:
+            out[k] = v
+    return out
+
+
+def connection(hostname, port, username, password, function, parameter):
+    url = 'http://{0}:{1}/sapcontrol?wsdl'.format(hostname, port)
+    client = Client(url, username=username, password=password)
+    _function = getattr(client.service, function)
+    if parameter is not None:
+        result = _function(parameter)
+    else:
+        result = _function()
+
+    return result
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            sysnr=dict(type='str', required=False),
+            port=dict(type='int', required=False),
+            username=dict(type='str', required=False),
+            password=dict(type='str', no_log=True, required=False),
+            hostname=dict(type='str', default="localhost"),
+            function=dict(type='str', required=True, choices=choices()),
+            parameter=dict(type='str', required=False),
+            force=dict(type='bool', default=False),
+        ),
+        required_one_of=[('sysnr', 'port')],
+        mutually_exclusive=[('sysnr', 'port')],
```

Comment:
so if I understood this right: Either one is required, but not both. Therefore both are defined with required=False and as such described above ?

Reply:
correct